spyder/plugins/editor/plugin.py

```diff
@@ -954,29 +954,70 @@ def get_plugin_actions(self):
         self.recent_file_menu = QMenu(_("Open &recent"), self)
         self.recent_file_menu.aboutToShow.connect(self.update_recent_file_menu)
 
-        file_menu_actions = [
+        from spyder.plugins.mainmenu.api import (
+            ApplicationMenus, FileMenuSections)
+        # New Section
+        self.main.mainmenu.add_item_to_application_menu(
             self.new_action,
-            MENU_SEPARATOR,
+            menu_id=ApplicationMenus.File,
+            section=FileMenuSections.New,
+            before_section=FileMenuSections.Restart)
+        # Open section
+        open_actions = [
             self.open_action,
             self.open_last_closed_action,
             self.recent_file_menu,
-            MENU_SEPARATOR,
-            MENU_SEPARATOR,
+            ]
+        for open_action in open_actions:
+            self.main.mainmenu.add_item_to_application_menu(
+                open_action,
+                menu_id=ApplicationMenus.File,
+                section=FileMenuSections.Open,
+                before_section=FileMenuSections.Restart)
+        # Save section
+        save_actions = [
             self.save_action,
             self.save_all_action,
             save_as_action,
             save_copy_as_action,
             self.revert_action,
-            MENU_SEPARATOR,
+            ]
+        for save_action in save_actions:
+            self.main.mainmenu.add_item_to_application_menu(
+                save_action,
+                menu_id=ApplicationMenus.File,
+                section=FileMenuSections.Save,
+                before_section=FileMenuSections.Restart)
+        # Print
+        print_actions = [
             print_preview_action,
             self.print_action,
-            MENU_SEPARATOR,
+            ]
+        for print_action in print_actions:
+            self.main.mainmenu.add_item_to_application_menu(
+                print_action,
+                menu_id=ApplicationMenus.File,
+                section=FileMenuSections.Print,
+                before_section=FileMenuSections.Restart)
+        # Close
+        close_actions = [
             self.close_action,
-            self.close_all_action,
-            MENU_SEPARATOR,
-        ]
+            self.close_all_action
+            ]
+        for close_action in close_actions:
+            self.main.mainmenu.add_item_to_application_menu(
+                close_action,
+                menu_id=ApplicationMenus.File,
+                section=FileMenuSections.Close,
+                before_section=FileMenuSections.Restart)
+        # Navigation
+        if sys.platform == 'darwin':
+            self.main.mainmenu.add_item_to_application_menu(
+                self.tab_navigation_actions,
+                menu_id=ApplicationMenus.File,
+                section=FileMenuSections.Navigation,
+                before_section=FileMenuSections.Restart)

```

Comment:
Why did you add this? I mean, it seems it was not necessary before.

Reply:
That code was in the `mainwindow` (it was doing `self.editor.tab_navigation_actions`) so I added it here since I was already touching this code